### PR TITLE
feat&fix: provides customization, and fix double indention

### LINKED ISSUE #14: vim 代码补全报错
- 创建 1.py
- 打开 1.py 并登录 fittencode，打印如下
```
Login successful, API key saved
```
- 输入
```
import torch 
```
- 在行尾编辑模式下输入 "\<C-L\>"，报错
- 错误如下
```
处理 function CodeCompletion 时发生错误:
第   77 行:
E964: 无效的列号：0
```

要额外进行配置吗

### PR DESCRIPTION
**Pull request by One-In-Dark**
Adds options for customization, like trigger keys. Adds vim help files. Options are described in help docs.

For advanced users, provides API for integration with other plugins.

BREAKING CHANGE: 'Accept()' has non-standard name.

'Accept()' is removed. Use 'FittenAccept()' instead.

======

bugfix: double indentation fixed

by setting 'paste' option to disable auto indent

Closes #14

### CODE REVIEW

File: plugin/fittencode.vim [hunk 1]
```diff
@@ -163,8 +163,11 @@ function! CodeCompletion()
...
 endfunction
 
-function! Accept()
-    echo "Accept"
+let g:fitten_accept_no_echo = 0
+function! FittenAcceptMain()
+    if g:fitten_accept_no_echo == 0
+        echo "Accept"
+    endif
     let default = pumvisible() ? "\<C-N>" : "\t"
```
Comment (luozhiya) [inline plugin/fittencode.vim:171]: We need different `default` values based on the different mapping keys. When there are no suggestions, `default` will insert the document. Please refer to the code on line 174.

File: plugin/fittencode.vim [hunk 2]
```diff
@@ -163,8 +163,11 @@ function! CodeCompletion()
     let b:fitten_suggestion = l:generated_text
 endfunction
 
-function! Accept()
-    echo "Accept"
+let g:fitten_accept_no_echo = 0
```
[the remark this answers is not in the document]
Comment (One-In-Dark) [reply plugin/fittencode.vim:166]: Thanks for the review! Somehow I think other log messages are necessary, while this is not the case. I may remove this option.

File: plugin/fittencode.vim (hunk 1 above)
Comment (One-In-Dark) [reply plugin/fittencode.vim:171]: @luozhiya

File: plugin/fittencode.vim (hunk 2 above)
[the remark this answers is not in the document]
Comment (One-In-Dark) [reply plugin/fittencode.vim]: @luozhiya

File: plugin/fittencode.vim (hunk 1 above)
Comment (luozhiya) [reply plugin/fittencode.vim:168]: > original keyboard input as default

We need to consider the special circumstances of all keys

> pum is an additional functionality

`pum` is necessary when we map to a `tab`

IMO, by providing two basic APIs, `FittenAccept` and `FittenAcceptable`, the interface can be more flexible, and the internal implementation can also be simplified. What's your opinion about this?

```vim
function! FittenAccept()
    echo "Accept"

    let l:text = b:fitten_suggestion

    call ClearCompletion()

    return l:text
endfunction

function! FittenAcceptable()
    return (mode() !~# '^[iR]' || !exists('b:fitten_suggestion')) ? 0 : 1
endfunction

function! FittenAcceptTab()
    if FittenAcceptable()
        return FittenAccept()
    else
        return pumvisible() ? "\<C-N>" : "\t"
    endif
endfunction

function! MapTab()
    inoremap <script><silent><nowait><expr> <Tab> FittenAcceptTab()
endfunction
```

**Pull request merged**